File: client/src/app/core/models/view-models/board-view.ts
import { EstimateType } from '@core/enums/estimate-type';
import { TaskPriority } from '@core/enums/task-priority';
import { SprintStatus } from '@core/enums/sprint-status';
import { StatusCategory } from '../status';
import { AppUser } from '../appuser';
import { BoardViewModel } from './board-view-model';

export interface BoardView {
  board: BoardViewModel;
  groups: BoardViewGroup[];
  users: AppUser[];
}

export interface BoardViewGroup {
  id: number;
  name: string;
  boardId: number;
  statusId: number | null;
  sortOrder: number;
  tasks: BoardViewTask[];
}

export interface BoardViewTask {
  id: number;
  assignees: AssigneeViewModel[];
  name: string;
  systemId: string;
  tags: string[];
  priority: TaskPriority | null;
  estimateType: EstimateType | null;
  estimateValue: number | null;
  dueDate?: string | null;
  createdAt: string;
  updatedAt: string;
  sprintId?: number | null;
  sprintName?: string | null;
  sprintStatus?: SprintStatus | null;
  sortOrder: number;
  statusId: number;
  statusName: string;
  statusKey: string;
  statusColor?: string | null;
  statusCategory: StatusCategory;
  projectId: number;
  workspaceId: number;
  workspaceKey: string;
}

export interface AssigneeViewModel {
  id: string;
  displayName: string;
  pictureUrl: string;
}

File: server/Netptune.Core/ViewModels/Boards/BoardView.cs
using Netptune.Core.Enums;
using Netptune.Core.ViewModels.Users;

namespace Netptune.Core.ViewModels.Boards;

public class BoardView
{
    public BoardViewModel Board { get; set; } = null!;

    public IEnumerable<BoardViewGroup> Groups { get; set; } = null!;

    public IEnumerable<UserViewModel> Users { get; set; } = null!;
}

public class BoardViewGroup
{
    public int Id { get; set; }

    public string Name { get; set; } = null!;

    public int BoardId { get; set; }

    public double SortOrder { get; set; }

    public int? StatusId { get; set; }

    public IList<BoardViewTask> Tasks { get; set; } = null!;
}

public class BoardViewTask
{
    public int Id { get; set; }

    public List<AssigneeViewModel> Assignees { get; set; } = null!;

    public string Name { get; set; } = null!;

    public string SystemId { get; set; } = null!;

    public int StatusId { get; set; }

    public string StatusName { get; set; } = null!;

    public string StatusKey { get; set; } = null!;

    public string? StatusColor { get; set; }

    public StatusCategory StatusCategory { get; set; }

    public IList<string> Tags { get; set; } = null!;

    public TaskPriority? Priority { get; set; }

    public EstimateType? EstimateType { get; set; }

    public decimal? EstimateValue { get; set; }

    public DateOnly? DueDate { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public double SortOrder { get; set; }

    public int ProjectId { get; set; }

    public int? SprintId { get; set; }

    public string? SprintName { get; set; }

    public SprintStatus? SprintStatus { get; set; }

    public int WorkspaceId { get; set; }

    public string WorkspaceKey { get; set; } = null!;
}
